src/plugin/index.ts

```diff
@@ -6,40 +6,59 @@ import chokidar from "chokidar";
 import path from "path";
 
 const packageName = "next-type-safe-routes";
-const typeFolder = path.join("@types", packageName);
+const defaultDesintaionFolder = path.join("@types", packageName);
 
 const log = (message: string) => {
   console.log(`\x1b[36m${packageName}\x1b[0m: ${message}`);
 };
 
-const writeTypesToDisc = (nextPagesDirectory: string) => {
-  // we assume the src directory is the directory containing the pages directory
-  const srcDir = path.dirname(nextPagesDirectory);
-  const typeFolderPath = path.join(srcDir, typeFolder);
-  const typeScriptFile = generateTypeScriptFile(nextPagesDirectory);
 
-  mkdirp.sync(typeFolderPath);
-  fs.writeFileSync(path.join(typeFolderPath, "index.d.ts"), typeScriptFile);
-
-  log(`types written to ${typeFolder}`);
+const writeTypesToDisc = ({ source, destination }: {
+  source: string;
+  destination: string;
+}) => {
+  const typeScriptFile = generateTypeScriptFile(source);
+  mkdirp.sync(destination);
+  fs.writeFileSync(path.join(destination, "index.d.ts"), typeScriptFile);
+  log(`types written to ${destination}`);
 };
 
-const run = (nextConfig: any = {}) => {
-  return Object.assign({}, nextConfig, {
-    webpack(config, options) {
-      // This seems to be the way to get the path to the pages
-      // directory in a Next.js app. Since it's possible to have a
-      // `/src` folder (https://nextjs.org/docs/advanced-features/src-directory)
-      // we cannot assume that it just in a `/pages` folder directly
-      // in the root of the project
-      const pagesDir = config.resolve.alias["private-next-pages"];
+const watchRoutes = ({ source, destination }: {
+  source: string;
+  destination: string;
+}) => {
       // Generate the types file when the app is being compiled
-      writeTypesToDisc(pagesDir);
+      writeTypesToDisc({
+         source: pagesDir,
```

Comment:
still referring to `pagesDir` although the variable doesn't exist, looks like something got lost in translation. patching it!